ts/Core/Series/Series.ts

```diff
@@ -4219,7 +4221,19 @@ class Series {
                 preserve.push(key + 'Data');
             });
 
-            if (options.data) {
+            let hasMapDataChanged;
+
+            // When updating the series.mapData, current value is required
+            // to be set before series.setData(), #11636
+            if (mapData) {
+                hasMapDataChanged = series.options.mapData !== mapData;
+
+                if (hasMapDataChanged) {
+                    series.options.mapData = mapData;
+                }
+            }
+
+            if (options.data || hasMapDataChanged) {
```

Comment:
We want to keep non-generic stuff out of the core Series.ts file. However there's a mechanism where we do a test for whether options have changed in order to determine `keepPoints`. I tried setting `keepPoints` to false, and that works for the use case. However I was expecting that `series.hasOptionChanged('mapData')` would be true, but it's not. So I would start with finding out why.

An alternative possible fix is the set [hasDerivedData to true](https://jsfiddle.net/highcharts/7raL89ep/) on the MapSeries prototype, but I am not sure of the consequences. In a sense it is semantically true because the data is composite and derived from `mapData` and `data`, but then again this flag is used with series where the data derived from another series.

Reply:
At the beginning I thought about the `series.hasOptionChanged` method too, but I don't think it's doing what we think. It's not comparing old option to an updated (new) option (during series.update()), but it's checking whether the option has been changed directly on the series level.

For example:
```js
series.update({ name: 'Test' });
// inside the series.update method
series.hasOptionChanged('name'); // -> false
```
but
```js
series.name = 'Test";
series.hasOptionChanged('name'); // -> true
```

The unit tests suggest the same. So I'm leaving this method as it is.

----------------------------------------------------------------------------
I'm also trying to fix it without setting `hasDerivedData: true` on the MapSeries prototype. Commit will be pushed soon

Reply:
Okay, so I misunderstood what it did.